esxi/datadog_checks/esxi/check.py

```diff
@@ -354,44 +395,15 @@ def set_version_metadata(self):
 
     def check(self, _):
         try:
-            context = ssl.SSLContext(protocol=ssl.PROTOCOL_TLS_CLIENT)
-            context.check_hostname = True if self.ssl_verify else False
-            context.verify_mode = ssl.CERT_REQUIRED if self.ssl_verify else ssl.CERT_NONE
-
-            if self.ssl_capath:
-                context.load_verify_locations(cafile=None, capath=self.ssl_capath, cadata=None)
-            elif self.ssl_cafile:
-                context.load_verify_locations(cafile=self.ssl_cafile, capath=None, cadata=None)
-            else:
-                context.load_default_certs(ssl.Purpose.SERVER_AUTH)
-
-            create_connection_method = socket.create_connection
-            if self.proxy_host:
-                socket.create_connection = lambda address, timeout, source_address, **kwargs: create_connection(
-                    address, timeout, source_address, self.proxy_host, self.proxy_port
-                )
-
-            connection = connect.SmartConnect(host=self.host, user=self.username, pwd=self.password, sslContext=context)
-            socket.create_connection = create_connection_method
-
-            self.conn = connection
-            self.content = connection.content
-
-            if self.content.about.apiType != "HostAgent":
-                raise Exception(
-                    f"{self.host} is not an ESXi host; please set the `host` config option to an ESXi host "
-                    "or use the vSphere integration to collect data from the vCenter",
-                )
-
-            self.log.info("Connected to ESXi host %s: %s", self.host, self.content.about.fullName)
-            self.count("host.can_connect", 1, tags=self.tags)
+            self.set_version_metadata()
+            self.gauge("host.can_connect", 1, tags=self.tags)
```

Comment:
If you move this line after the `try/except` block and remove it from `initiate_api_connection` method, you'll report only once. Does it make sense?

Reply:
Discussed in zoom